### PR TITLE
Run Rancher tests cases in parallel.

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

Run Rancher tests cases in parallel.

### Motivation

Speed up tests.

### More

- [x] Added/updated tests

### CODE REVIEW

File: provider/rancher/rancher_test.go [hunk 1]
```diff
@@ -104,16 +103,19 @@ func TestRancherServiceFilter(t *testing.T) {
...
 	}
 
-	for _, e := range services {
-		actual := provider.serviceFilter(e.service)
-		if actual != e.expected {
-			t.Fatalf("expected %t, got %t", e.expected, actual)
-		}
+	for _, test := range testCases {
+		test := test
+		t.Run("", func(t *testing.T) {
```
Comment (timoreimann) [inline provider/rancher/rancher_test.go]: Can we use a non-empty first parameter for the test description?

(It's missing elsewhere too.)

Comment (ldez) [reply provider/rancher/rancher_test.go]: It's hard to named tests that I don't created. 😉

Comment (timoreimann) [reply provider/rancher/rancher_test.go]: How about using indices at least? It's far from ideal but better than nothing.

Comment (ldez) [reply provider/rancher/rancher_test.go]: Done!

**Pull request merged**